presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushAggregationThroughOuterJoin.java

```diff
@@ -120,7 +122,8 @@ public Result apply(AggregationNode aggregation, Captures captures, Context cont
         if (join.getFilter().isPresent()
                 || !(join.getType() == JoinNode.Type.LEFT || join.getType() == JoinNode.Type.RIGHT)
                 || !groupsOnAllColumns(aggregation, getOuterTable(join).getOutputSymbols())
-                || !isDistinct(context.getLookup().resolve(getOuterTable(join)), context.getLookup()::resolve)) {
+                || !isDistinct(context.getLookup().resolve(getOuterTable(join)), context.getLookup()::resolve)
+                || !isUsingSymbols(aggregation, getInnerTable(join))) {
```

Comment:
>We want all aggregations to be from outer side exclusively, no?

I'm not sure if the naming here is correct. I just followed the convention. It says "inner table" when referring to the right table of LEFT join.
We want to push the AggregationNode into the right source of LEFT join or into the left source of RIGHT join.

>Also, didn't previous code deal with mixed aggregations?

It seemed so, because it only required that _some_ of the Aggregation symbols matched the source. But it failed on mixed aggregations (vide the test cases I added). 

This PR is just a fix. Further we could extend the rule to remap Aggregations by equi-clauses before pushdown.

Reply:
> I'm not sure if the naming here is correct. I just followed the convention. It says "inner table" when referring to the right table of LEFT join.

Right, I confused the names. Thanks